### ISSUE TITLE
Unable to upload Zarrs to staging because "There were headers present in the request which were not signed"

### ISSUE THREAD

**Issue by jwodder** [dandi/dandi-archive#1494]
Currently, attempting to upload a Zarr to staging fails when PUTing a file to S3 with a 403 error and a response of the form:

```xml
<Error>
  <Code>AccessDenied</Code>
  <Message>There were headers present in the request which were not signed</Message>
  <HeadersNotSigned>content-md5</HeadersNotSigned>
  <RequestId>S086HFMAEH43CNF6</RequestId>
  <HostId>9Uenhku5cje+7L4e4+h3y0aBwv4CsC4+d/01j/57xuBTeF5RdGtJ8jTtThUFAJ6EpcsjXTMXLi8=</HostId>
</Error>
```

Upload succeeds when testing against a Dockerized environment.  I have not tried uploading to production.

**Comment by waxlamp** [comment]
@AlmightyYakob, can you look into this?

In case it helps, I found [this SO question](https://stackoverflow.com/questions/45653850/there-were-headers-present-in-the-request-which-were-not-signed) but it is a few years old and I don't know if it applies to our situation.

**Comment by AlmightyYakob** [comment]
@danlamanna and I discussed this offline, and he's going to be taking it on.

**Comment by jwodder** [comment]
Resolved.

**Comment by AlmightyYakob** [comment]
Resolved via #1497, #1498 and https://github.com/dandi/dandi-cli/pull/1215
